inngest/src/result.rs

```diff
@@ -36,24 +36,97 @@ impl IntoResponse for SdkResponse {
 }
 
 #[derive(Debug)]
-pub enum InngestError {
+pub enum SimpleError {
+    /// A catch-all error type for business logic errors
     Basic(String),
     RetryAt(RetryAfterError),
     NoRetry(NonRetryableError),
 
-    // Used for invoked functions that don't have a response
+    /// Used for invoked functions that don't have a response
     NoInvokeFunctionResponseError,
+}
 
-    // These are not expected to be used by users
-    #[allow(private_interfaces)]
+#[derive(Debug)]
+pub enum InngestError {
+    Simple(SimpleError),
+
+    /// Internal only. These are not expected to be used by users. These must be propagated to their callers
     Interrupt(FlowControlError),
 }
 
+impl From<SimpleError> for InngestError {
+    fn from(err: SimpleError) -> Self {
+        InngestError::Simple(err)
+    }
+}
+
+/// Create a basic error using format! syntax
+#[macro_export]
+macro_rules! basic_error {
+    ($($arg:tt)*) => {
+        $crate::result::InngestError::Simple($crate::result::SimpleError::Basic(
+            format!($($arg)*),
+        ))
+    };
+}
+
+/// Correctly propagate the flow control error while providing the user with a simple error
+#[macro_export]
+macro_rules! simplify_err {
```

Comment:
This, combined with the `SimpleError` was my idea for allowing users to throw away (and return) the internal control flow error they don't need to worry about. Not sure if this really hits the way I thought it would. It's not horrible.

Reply:
huh, I see! this is smart!

Reply:
I'm down on the logic, just thinking if there's a better name for the macro itself.

Ideally something that works well with the keyword `run`.

Response from chatgpt. :thinking: 
![image](https://github.com/user-attachments/assets/dd51d4eb-3bae-4b5e-b8a1-ebab3f7908ca)

Reply:
The macro can/should be used anywhere the user wants to inspect the `Error` type that is returned for some dev facing error, effectively grabbing the `DevResult`. So it's not really limited to the step.run use case. 

I've renamed the macro to that effect. Lmk how it feels